### DISCUSSION TITLE
Support toggling code completions (copilot) on/off with a keybind

### DISCUSSION THREAD

**Discussion by brandonzylstra** [zed-industries/zed#23971] [Feature Requests]
### Summary

The documentation (and comments in various files) give information such as this:

## Show Copilot Suggestions ##

- Description: Whether or not to show Copilot suggestions as you type or wait for a `copilot::Toggle`.
- Setting: `show_copilot_suggestions`
- Default: `true`

### Options ###

`boolean` values

---

and yet nowhere (that I can find) is `copilot::Toggle` defined.  Is this a keystroke that I can add to my settings or keymap file?

I would like to avoid having Copilot spew suggestions while I'm in the middle of typing code, but I want to be able to tell it to make suggestions when I want them.  As far as I can tell, this is not possible with the current version of Zed.  

If it _is_ possible, I would love to know how to do it.

**Comment by brandonzylstra** [comment]
Poking around in the documentation for the `keymap.json` file I noticed that `namespace::Verb` is a ubiquitous pattern, so I thought I had found the answer, however the following returned an error ("Value not accepted" followed by a list of valid values), so I think `copilot::Toggle` might be a mistake in the documentation.

```
[
  {
    "context": "Editor",
    "bindings": {
      "ctrl-shift-I": "copilot::Toggle"
    }
  }
]
```

I tried substituting a valid value, but it still doesn't work:

```
[
  {
    "context": "Editor",
    "bindings": {
      "ctrl-shift-I": "copilot::Suggest"
    }
  }
]
```

**Comment by JosephTLyons** [comment]
I'm not entirely sure that there's not some way to fiddle with the keybindings.json file to get this to work, but my gut instinct is that we currently can't. Happy to hear I'm wrong, if someone on the team knows differently.

**Comment by bofm** [comment]
+1 to the topic. It wasn't easy to find this issue. Maybe we could change the issue title somehow? I was searching for something like "zed copilot disable auto-suggestion".

**Comment by brandonzylstra** [comment]
> +1 to the topic. It wasn't easy to find this issue. Maybe we could change the issue title somehow? I was searching for something like "zed copilot disable auto-suggestion".

I edited the title... it's a little bit long but maybe it will be easier to find now.

**Comment by sebastiansandqvist** [comment]
It looks like `copilot::toggle` used to exist (https://github.com/zed-industries/zed/commit/ebd06b43f5deda4b4a09d11514e0baac4c0f9673) and that as of PR https://github.com/zed-industries/zed/pull/2316 it was possible to toggle on/off with a keybinding. Unsure when this support was removed, but I'm hoping this context helps restore that functionality since enabling copilot on-demand was something I was also trying to do.

**Comment by probably-neb** [comment]
Hey! Thanks for reporting the confusion here. Definitely seems like this is something you should be able to do in Zed. 
We've decided to move feature requests like this to discussions so I'm going to move it there now. 
Feel free to keep voicing your desire to have this in Zed, the more people we know want to have something like this the more likely we are to prioritize it
